File: common/src/main/clojure/lukebemish/clojuremixin/api/platform.clj
(ns lukebemish.clojuremixin.platform
  (:import lukebemish.clojuremixin.Platform))

(defn get-platform [] (Platform/getPlatform))

(defmacro defn-specific [name]
  (list `do
        (list `use (list `quote (symbol (str *ns* (.name (Platform/getPlatform)))))
        (list `defn name (symbol (str name "-impl"))))))

File: common/src/main/java/lukebemish/clojuremixin/Platform.java
package lukebemish.clojuremixin;

import dev.architectury.injectables.annotations.ExpectPlatform;

public enum Platform {
    FORGE("forge"),
    FABRIC("fabric");

    private final String name;

    @ExpectPlatform
    public static Platform getPlatform() {
        throw new AssertionError();
    }

    Platform(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }
}
